### DISCUSSION TITLE
What significance do the constants in multiplier formula for score calculation have?

### DISCUSSION THREAD

**Discussion by FL44ME** [iSoron/uhabits#1112] [Technical Discussion] [answered]
https://github.com/iSoron/uhabits/blob/aee0da2c649aae5bfae38e587858a3e34fc5414e/android/uhabits-core/src/main/java/org/isoron/uhabits/core/models/Score.java#L66

Do the constants 0.5 and 13.0 have any practical significance in the multiplier formula or are these arbitrary smoothing factors?

@iSoron @hiqua

**Comment by hiqua** [comment, marked as answer]
It's basically one constant: `pow(0.5, 1/13.0)` to the power of `frequency`.

I think it's arbitrary, in this case it's around 0.95, so if `frequency` is 1, the old score should account for 95% of the new score.

See also https://github.com/iSoron/uhabits/discussions/689 ("How is the score calculated?").

**Reply by FL44ME** [reply]
Thanks. As a follow-up: I feel that this method used for numerical habits should also have a reverse target feature (such as when trying to limit something). While for most bad habits that one would like to completely end, a 0 for target value works, something like weight is dynamic and people could be trying to both gain or lose weight. While reversing the structure of the question is a simple fix, a consistent pattern (higher score = better) on the dashboard across all habits would allow the user to easily get a feel of their scores quickly without having to pay particular attention to different habits. Is this feasible?

**Reply by hiqua** [reply]
Sounds like you're talking about https://github.com/iSoron/uhabits/pull/1101

**Reply by FL44ME** [reply]
> Sounds like you're talking about #1101

Thanks for pointing it out

**Reply by iSoron** [reply]
This answer is correct. That particular constant was chosen so that the score reaches 80% after one month, 96% after two months and 99% after three months. I though about making this configurable, but there was very little user demand for it.
